R/utils.R

```diff
@@ -0,0 +1,32 @@
+fromRJSON <- function(x, ...) {
+  return(jsonlite::fromJSON(sprintf(x, ...),
+    simplifyVector = FALSE,
+    simplifyDataFrame = FALSE,
+    simplifyMatrix = FALSE
+  ))
+}
+
+handle_err_result <- function(res) {
+  if (res$type == "error") {
+    return(TRUE)
+  }
+  return(FALSE)
+}
+
+next_id <- 0
+get_new_id <- function() {
+  # next_id <<- next_id + 1
+  return(next_id)
```

Comment:
Doesn't this just stay at 0 always?

Reply:
Ah lol, yeah, I did not mean to commit and push this change :D

Reply:
In general, this should be written as a function factory, to keep the id local in scope, consider something like:

```R
make_my_amazing_factory <- function() {
   counter <- 0;
   function() counter <<- counter + 1
}

count <- make_my_amazing_factory()
```

Reply:
See: https://adv-r.hadley.nz/function-factories.html#stateful-funs